mirgecom/materials/tacot.py

```diff
@@ -0,0 +1,527 @@
+""":mod:`~mirgecom.materials.tacot` evaluates TACOT-related data.
+
+    TACOT is the Theoretical Ablative Composite for Open Testing, a surrogate
+    composite material closely related to PICA (Phenolic Impregnated Carbon
+    Ablator) proposed as an open-source material for the Ablation Workshop.
+
+.. autoclass:: Pyrolysis
+.. autoclass:: SolidProperties
+.. autoclass:: GasProperties
+.. autoclass:: BprimeTable
+.. autoclass:: WallTabulatedEOS
+
+Helper Functions
+----------------
+.. autofunction:: eval_spline
+.. autofunction:: eval_spline_derivative
+"""
+
+__copyright__ = """
+Copyright (C) 2023 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+import scipy  # type: ignore[import]
+from scipy.interpolate import CubicSpline  # type: ignore[import]
+from meshmode.dof_array import DOFArray
+from pytools.obj_array import make_obj_array
+from mirgecom.fluid import ConservedVars
+from mirgecom.wall_model import WallDependentVars, WallEOS
+
+
+class BprimeTable():
+    """Class containing the table for energy balance at the surface.
+
+    This class is only required for uncoupled cases, where only the wall
+    portion is evaluated. This is NOT used for fully-coupled cases.
+    """
+
+    # FIXME read from the "materials" folder to not get explictly the table
```

Comment:
Still planning to do this?

Reply:
Fixed in [efa834b](https://github.com/illinois-ceesd/mirgecom/pull/875/commits/efa834b1505237ddead31d538b704996717109a3)